src/main/java/net/coderbot/iris/gl/program/ProgramUniforms.java

```diff
@@ -155,8 +160,14 @@ public OptionalInt location(String name, UniformType type) {
 				return OptionalInt.empty();
 			}
 
-			locations.put(id, name);
-			uniformNames.put(name, type);
+			if (!locations.containsKey(id) && !uniformNames.containsKey(name)) {
```

Comment:
The only purpose of externally managed uniforms is just to suppress warnings for uniforms not managed through the uniform system - there isn't a pressing need to check them, but if we do, we should throw a hard error instead. Not sure if we should check if a normal uniform collides with an externally managed uniform.